src/solidity/stateDecoder.js

```diff
@@ -0,0 +1,79 @@
+var decoder = require('./stateDecoder')
+var astHelper = require('./astHelper')
+var typeDecoder = require('./typeDecoder')
+
+/**
+  * decode the contract state storage
+  *
+  * @param {Array} storage location  - location of all state variables
+  * @param {Map} storageContent  - storage
+  * @return {Map} - decoded state variable
+  */
+function decodeState (storageLocations, storageContent) {
+  if (storageContent['0x']) {
+    storageContent['0x00'] = storageContent['0x']
+    storageContent['0x'] = undefined
+  }
+  var ret = {}
+  for (var k in storageLocations) {
+    var stateVar = storageLocations[k]
+    var location = {
+      slot: stateVar.location.slot,
+      offset: stateVar.location.offset
+    }
+    ret[stateVar.name] = decoder.decode(stateVar.type, storageContent, location)
+  }
+  return ret
+}
+
+/**
+  * return all storage location variables of the given @arg contractName
+  *
+  * @param {String} contractName  - name of the contract
+  * @param {Object} astList  - AST nodes of all the sources
+  * @return {Object} - return the location of all contract variables in the storage
+  */
+function getStateVariableLocations (contractName, astList) {
+  var stateDefinitions = astHelper.getStateDefinition(astList, contractName)
+  var ret = []
+  if (!stateDefinitions) {
+    return ret
+  }
+  var location = {
+    offset: 0,
+    slot: 0
+  }
+  for (var k in stateDefinitions) {
+    var variable = stateDefinitions[k]
+    if (variable.name === 'VariableDeclaration') {
```

Comment:
ok
I took a look also on the AST, There might be a misbehavior somewhere:
with this contract:
`
contract Test {
    int stateVar = 35;
}
`
I got the following VariableDeclaration:
`
"{"attributes":{"name":"stateVar","type":"int256"},"children":[{"attributes":{"name":"int"},"id":5602188,"name":"ElementaryTypeName","src":"20:3:0"},{"attributes":{"hexvalue":"3335","subdenomination":null,"token":null,"type":"int_const 35","value":"35"},"id":5602652,"name":"Literal","src":"35:2:0"}],"id":5602708,"name":"VariableDeclaration","src":"20:17:0"}"
`

Litteral is defined and contains ´int_const´
I got the same AST node as `úint constant stateVar = 23`